detox/e2e/test/smoke_test/channels.e2e.js

```diff
@@ -0,0 +1,151 @@
+// - [#] indicates a test step (e.g. # Go to a screen)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element testID when selecting an element. Create one if none.
+// *******************************************************************
+
+import {MainSidebar} from '@support/ui/component';
+import {ChannelScreen} from '@support/ui/screen';
+import {
+    Channel,
+    Post,
+    Preference,
+    Setup,
+    Team,
+    User,
+} from '@support/server_api';
+
+describe('Channels', () => {
+    let testMessage;
+    let unreadChannel;
+    let favoriteChannel;
+    let publicChannel;
+    let privateChannel;
+    let nonJoinedChannel;
+    let directMessageChannel;
+    let dmOtherUser;
+    let nonDmOtherUser;
+    const {
+        channelNavBarTitle,
+        closeMainSidebar,
+        openMainSidebar,
+    } = ChannelScreen;
+    const {
+        getChannelByDisplayName,
+        getFilteredChannelByDisplayName,
+        hasChannelDisplayNameAtIndex,
+        hasFilteredChannelDisplayNameAtIndex,
+        searchInput,
+    } = MainSidebar;
+
+    beforeAll(async () => {
+        const {user, channel, team} = await Setup.apiInit();
+        unreadChannel = channel;
+        testMessage = `Mention @${user.username}`;
+        await Post.apiCreatePost({
+            channelId: unreadChannel.id,
+            message: testMessage,
+        });
+
+        ({channel: favoriteChannel} = await Channel.apiCreateChannel({type: 'O', prefix: '4-favorite-channel', teamId: team.id}));
+        await Channel.apiAddUserToChannel(user.id, favoriteChannel.id);
+        await Preference.apiSaveFavoriteChannelPreference(user.id, favoriteChannel.id);
+
+        ({channel: publicChannel} = await Channel.apiCreateChannel({type: 'O', prefix: '3-public-channel', teamId: team.id}));
+        await Channel.apiAddUserToChannel(user.id, publicChannel.id);
+
+        ({channel: privateChannel} = await Channel.apiCreateChannel({type: 'P', prefix: '2-private-channel', teamId: team.id}));
+        await Channel.apiAddUserToChannel(user.id, privateChannel.id);
+
+        ({channel: nonJoinedChannel} = await Channel.apiCreateChannel({type: 'O', prefix: '1-non-joined-channel', teamId: team.id}));
+
+        ({user: dmOtherUser} = await User.apiCreateUser({prefix: 'testchannel-1'}));
+        ({channel: directMessageChannel} = await Channel.apiCreateDirectChannel([user.id, dmOtherUser.id]));
+        await Post.apiCreatePost({
+            channelId: directMessageChannel.id,
+            message: testMessage,
+        });
+
+        ({user: nonDmOtherUser} = await User.apiCreateUser({prefix: 'testchannel-2'}));
+        await Team.apiAddUserToTeam(nonDmOtherUser.id, team.id);
+
+        // # Open channel screen
+        await ChannelScreen.open(user);
+    });
+
+    afterAll(async () => {
+        await ChannelScreen.logout();
+    });
+
+    it('MM-T3184 should display unfiltered channels list', async () => {
```

Comment:
The test case mentioned about "Rotate the device to put display in landscape mode". Can't see if implemented but should it be considered here as well?

Reply:
@saturninoabril Actually landscape mode is now obsolete. I'll update the TM4J. Thanks for catching!